uefi-raw/src/protocol/misc.rs

```diff
@@ -22,3 +23,30 @@ pub struct TimestampProperties {
     /// example, a 24-bit counter would have an end value of `0xff_ffff`.
     pub end_value: u64,
 }
+
+/// Properties of Reset Notification.
+#[derive(Debug)]
+#[repr(C)]
+pub struct ResetNotificationProtocol {
+    pub register_reset_notify: unsafe extern "efiapi" fn(
+        this: *const Self,
```

Comment:
nit: let's make the `Self` parameters `*mut`. These are "logically" mutating operation (adding / removing), regardless of whether the implementation actually requires mutation.